tests/unit/encoding.ts

```diff
@@ -0,0 +1,235 @@
+import registerSuite = require('intern!object');
+import assert = require('intern/chai!assert');
+import {Ascii, Utf8, Hex, Base64} from 'src/encoding';
+
+const ENCODED_STRING = 'the catｷ and ｦ the hat';
+const ASCII_BUFFER = [116, 104, 101, 32, 99, 97, 116, 65399, 32, 97, 110, 100, 32, 65382, 32, 116, 104, 101, 32, 104, 97, 116];
+const UTF8_BUFFER = [116, 104, 101, 32, 99, 97, 116, 239, 189, 183, 32, 97, 110, 100, 32, 239, 189, 166, 32, 116, 104, 101, 32, 104, 97, 116];
+const HEX_BUFFER = [0xCB, 0xDE];
+const BASE64_BUFFER = [83, 71, 86, 115, 98, 71, 56, 103, 86, 71, 104, 108, 99, 109, 85];
+
+const UTF8_SURROGATE_TEST_DATA = [
+	{
+		encoding: '\uD800',
+		decoding: [0xED, 0xA0, 0x80]
+	},
+	{
+		encoding: '\uD800\uD800',
+		decoding: [0xED, 0xA0, 0x80, 0xED, 0xA0, 0x80]
+	},
+	{
+		encoding: '\uD80A',
+		decoding: [0xED, 0xA0, 0x080A]
+	},
+	{
+		encoding: '\uD800\uD834\uDF06\uD800',
+		decoding: [0xED, 0xA0, 0x80, 0xF0, 0x9D, 0x86, 0xED, 0xA0, 0x80]
+	},
+	{
+		encoding: '\uD9AF',
+		decoding: [0xED, 0xA6, 0xAF]
+	},
+	{
+		encoding: '\uDBFF',
+		decoding: [0xED, 0xAF, 0xBF]
+	},
+	{
+		encoding: '\uDC00',
+		decoding: [0xED, 0xB0, 0x80]
+	},
+	{
+		encoding: '\uDC00\uDC00',
+		decoding: [0xED, 0xB0, 0x80, 0xED, 0xB0, 0x80]
+	},
+	{
+		encoding: '\uDC00A',
+		decoding: [0xED, 0xB0, 0x080A]
+	},
+	{
+		encoding: '\uDC00\uD834\uDF06\uDC00',
+		decoding: [0xED, 0xB0, 0x80, 0xF9, 0x9D, 0x8C, 0xED, 0xB0, 0x80]
+	},
+	{
+		encoding: '\uDEEE',
+		decoding: [0xED, 0xBB, 0xAE]
+	},
+	{
+		encoding: '\uDFFF',
+		decoding: [0xED, 0xBF, 0xBF]
+	}
+];
+
+registerSuite({
+	name: 'Encoding Classes',
+
+	'Ascii': {
+		'.encode()'() {
+			let buffer = Ascii.encode(ENCODED_STRING);
+			assert.strictEqual(buffer.length, ENCODED_STRING.length);
+
+			for (let i = 0, length = buffer.length; i < length; i++) {
+				assert.strictEqual(buffer[i], ASCII_BUFFER[i]);
+			}
```

Comment:
Since we're outputting number arrays all the time now, the test could just use assert.deepEqual instead of checking the length and individual character values. Also, there's an assert.lengthOf for checking the length of things.